purchase_minimum_amount/tests/test_po_minimum_amount_block.py

```diff
@@ -0,0 +1,126 @@
+        user =\
+            self.users_obj.with_context({'no_reset_password': True}).\
+            create({
+                'name': 'Purchase User',
+                'login': login,
+                'password': 'test',
+                'email': 'test@yourcompany.com',
+                'company_id': company.id,
+                'company_ids': [(4, company.id)],
+                'groups_id': [(6, 0, group_ids)],
+            })
+        return user.id
+
+    def _create_purchase(self, line_products):
+        """ Create a purchase order.
+        ``line_products`` is a list of tuple [(product, qty)]
+        """
+        lines = []
+        for product, qty in line_products:
+            line_values = {
+                'name': product.name,
+                'product_id': product.id,
+                'product_qty': qty,
+                'product_uom': product.uom_id.id,
+                'price_unit': 100,
+                'date_planned': time.strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+            }
+            lines.append((0, 0, line_values))
+        purchase = self.po_obj.create({
+            'partner_id': self.partner1.id,
+            'approval_block_id': self.po_block_obj.id,
+            'order_line': lines,
+            'company_id': self.company1.id,
+        })
+        return purchase
+
+    def test_po_amount_block_1(self):
+        "Test PO Block for Minimum Threshold Vendor Amount"
+        self.partner1.write({'minimum_po_amount': 1500.00})
+
+        # Create a PO with an amount below the minimum and check that the
+        # Approval Block Reason is correctly assign
+        purchase1 = self._create_purchase(
+            [(self.product1, 1),
+             (self.product2, 5),
+             (self.product3, 8),
+             ]
+        )
+
+        self.assertEquals(
+            purchase1.approval_block_id,
+            self.env.ref(
+                'purchase_minimum_amount.minimum_amount_block_reason'))
+
+        # Release the PO by pressing the button and then confirming the order
+        purchase1.sudo(self.user2_id).button_release_approval_block()
+        purchase1.sudo().button_confirm()
+        self.assertEquals(purchase1.state, 'purchase')
+
+    def test_po_amount_block_2(self):
+        "Test PO Block for Minimum Threshold Vendor Amount"
+        self.partner1.write({'minimum_po_amount': 1500.00})
+
+        purchase1 = self._create_purchase(
+            [(self.product1, 1),
+             (self.product2, 5),
+             (self.product3, 8)])
+
+        self.assertEquals(
+            purchase1.approval_block_id,
+            self.env.ref(
+                'purchase_minimum_amount.minimum_amount_block_reason'))
+
+        for po_line in purchase1.order_line:
+            if po_line.product_id == self.product1:
+                po_line.product_qty = 10
+
+        self.assertEquals(
+            purchase1.approval_block_id,
+            self.env['purchase.approval.block.reason'])
+
```

Comment:
This is not the way to solve the issue. Restore the user and check which permissions are needed and add it the user.